### PR TITLE
Update deps

### PR DESCRIPTION
**Pull request by DuongTyler**
object-path update
https://npmjs.com/advisories/1573

### CODE REVIEW

**Comment by DuongTyler** [comment]
ok don't merge this, apparently somehow I got it working one time, but every other time I try running npm start, it fails to build. Apparently typescript is broken on react >4.0.0

File: frontend/src/components/headerSection/header.tsx [hunk 1]
```diff
@@ -24,14 +24,34 @@ interface HeaderProps
```
Comment (zapplecat) [inline frontend/src/components/headerSection/header.tsx:24]: Did you rebase off my branch or something here? Doesn't seem part of the deps change.

File: frontend/package.json [hunk 2]
```diff
@@ -5,25 +5,25 @@
...
+    "react-dom": "^17.0.1",
     "react-emoji-render": "^1.2.4",
-    "react-in-viewport": "^1.0.0-alpha.14",
+    "react-in-viewport": "^1.0.0-alpha.15",
     "react-router-dom": "^5.2.0",
-    "react-scripts": "3.4.3",
+    "react-scripts": "^4.0.0",
     "react-visibility-sensor": "^5.1.1",
-    "typescript": "^3.9.7"
+    "typescript": "^4.0.5"
```
Comment (zapplecat) [inline frontend/package.json]: We should keep an eye on https://github.com/facebook/create-react-app/issues/10110. I've subbed to that thread; maybe we should have an issue for it...?

**Review by zapplecat — APPROVED** [review]

File: frontend/package.json (hunk 2 above)
Comment (zapplecat) [reply frontend/package.json]: We should change this from `^` to `~`

**Comment by zapplecat** [comment]
lgtm

**Pull request merged**